client/state/terms/selectors.js

```diff
@@ -103,7 +103,16 @@ export const getTermsHierarchyForQueryIgnoringPage = createSelector(
 			return terms;
 		}
 
-		return ( new TreeConvert( 'ID' ) ).treeify( terms );
+		// For each site term object, add `parent` and `order` properties.
+		// These are used by the TreeConvert library ( which mutates ) to build the hierarchy.
+		const treeReadyTerms = terms.map( ( term, i ) => {
+			return Object.assign( {}, term, {
```

Comment:
Fancier option now with Babel 6 is [object spread properties](https://github.com/sebmarkbage/ecmascript-rest-spread) (stage 2 proposal):

```js
return {
	...term,
	parent: term.parent || 0,
	order: i
};
```